### DISCUSSION TITLE
Origin of Interlisp iterative statement and Common Lisp LOOP macro

### DISCUSSION THREAD

**Discussion by masinter** [Interlisp/medley#1592] [Show and tell]
A listing from 1972 Lisp 360 features I had implemented, 18 pp.
Now available at https://archive.computerhistory.org/resources/access/text/2024/03/102721861-05-01-acc.pdf#page=9 
![image](https://github.com/Interlisp/medley/assets/1116587/5dfa545d-d399-47ce-999d-57fdcac138b3)
https://www.computerhistory.org/collections/catalog/102721861

**Comment by RyanBurnside** [comment]
I think it's worth a thread bump to congratulate you on some awesome work.
For some reason, there is a movement (especially by newcomers from Haskell etc) to try and paint Lisp as a single paradigm Functional only language with no looping and only recursion but people are missing a LOT of history. There is a lot of looping, goto, and setting values. I've always been a mixed paradigm man and it's nice to see the roots of Interlisp's Iterative Statement.
